app/views/layouts/application.html.erb

```diff
@@ -42,6 +42,14 @@
             <% if current_user %>
               <li><%= link_to "My Trainings", user_trainings_path(current_user) %></li>
             <% end %>
+            <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Places <span class="caret"></span></a>
+            <ul class="dropdown-menu">
+              <li><%= link_to "Places", places_path %></li>
+                <% if current_user %>
```

Comment:
this condition will always evaluate to `true` as we introduced the NullObject pattern in your project, I think we should use `user_signed_in?` method here

Reply:
OK